### DISCUSSION TITLE
Rotary Switch ?

### DISCUSSION THREAD

**Discussion by jongena** [hneemann/Digital#980] [Q&A] [answered]
Would it be possible to include a rotary switch in the switch menu ? Or is there a suitable alternative that could be used with the existing components ?

**Comment by mengstr** [comment]
Being digital logic I assume you'd either want the rotary switch to either select one of many inputs and sent it to  single/common output, or the other way - you have one input that you want to send to one selectable output.

That's simply a multiplexer or demultiplexer and are available as ready made components under Plexers.

By setting a number to the input on the bottom (or top) the multiplexer "rotates rotary switch" inside the component.
<img width="963" alt="Screenshot 2022-07-11 at 23 53 33" src="https://user-images.githubusercontent.com/325326/178365884-0f21c155-b4c3-4420-ac72-99af6791c08b.png">

**Reply by prlaba** [reply]
It would be cool to come up with some sort of Rotary Encoder component.  It would appear as a disk that you would click on along its outside edge and rotate either clockwise or counterclockwise.  The number of bits assigned determines its maximum positive and negative value.  Its output is a signed n-bit value, with negative values indicating counterclockwise rotation and positive values indicating clockwise rotation.  You could come up with a "granularity" attribute that indicates the number of rotations to reach its maximum value (default 1).

**Reply by hneemann** [reply]
But there is a rotary encoder available. Why not use this one?

**Reply by prlaba** [reply]
Because I'm an idiot.🥴  Sorry I missed it!

**Reply by hneemann** [reply]
Nevermind 😄

**Comment by jongena** [comment, marked as answer]
Many thanks for all the quick and very useful answers !
